### ISSUE TITLE
[BartModel] Question for BartModel Output Shape when I pass the 'decoder_input_ids'

### ISSUE THREAD

**Issue by Bannng** [huggingface/transformers#5853]
# ❓ Questions & Help

## Details

Hi I'm trying to using model named 'BartForConditionalGeneration' for text generation, 
The main question and problem is, when I pass the 'decoder_input_ids'  to the model as the input of forward in model,
the first index of model's output shows the tensor shape as [batch_size, 1 , vocab_size] all the time.
I might misunderstood the model and Docs of Bart overall,
when I only pass the Input_ids without decoder_input_ids, it worked right.
but It never make the shape of tensor like [batch_size, target_seq_len, vocab_size] when I put 'decoder_input_ids'

I think when I put the decoder_input_ids as [batch_size, target_seq_len], the Model's output should be like 
[batch_size, target_seq_len, vocab_size] (for prediction scores)

the reason why I tried to pass the decoder_input_ids was , I think Bart is seq2seq model, and their output logits will be computed just like the basic transformer, so I wanted to pretrain this model.

Please help me to solve this problem.  If I misunderstood the concept of Bart at all, just comment me. 
I need a clean view for my situation 

I used mBartTokenizer and also 'facebook/mbart-large-en-ro'  pretrained weights, 
I used Model BartForConditionalGeneration.from_pretrained('facebook/mbart-large-en-ro')
It also didn't worked for normal BartModel (with same pretrained config)

I'm really confused about it so the whole question looks like a mass, Sorry for that
Thank you for reading my questions. Please Help Me.
![1](https://user-images.githubusercontent.com/51171232/87811751-1a4d1700-c89a-11ea-8e47-0ae2987dfe29.PNG)
